cpp/src/strings/case.cu

```diff
@@ -167,7 +168,10 @@ struct count_bytes_fn {
       size += converter.process_character(u8);
     }
     // this is every so slightly faster than using the cub::warp_reduce
-    if (size > 0) atomicAdd(d_offsets + str_idx, size);
+    if (size > 0) {
+      cuda::atomic_ref<size_type, cuda::thread_scope_block> ref{*(d_offsets + str_idx)};
```

Comment:
Is `thread_scope_block` correct?

Reply:
I think that might be correct, but it'd be good to double check with @davidwendt.

In my understanding this is safe because only one block (really one warp, but that doesn't seem to be an atomic scope in libcudacxx) is accessing the data at `d_offsets + str_idx`. If there is a way to express atomicity at the warp level in libcudacxx I'd like to know how. Warps are not mentioned in https://nvidia.github.io/libcudacxx/extended_api/memory_model.html.

Similar reasoning applies to the other places where you left this comment.

Reply:
This is correct. No two warps will update the same memory location so thread-scope should be sufficient here.